docs/ASDK-2.0-Layout-Documention.md

```diff
@@ -0,0 +1,802 @@
+
+Replaced layoutables can have intrinsic dimensions—width and height values that are defined by the layoutable itself, rather than by its surroundings. For example, if an `ASImageNode` has a width set to `ASDimensionAuto`, the width of the linked image file will be used. For an `ASImageNode` the intrinsic content size would be calculated based on the text content.
+
+## Height and Width
+
+### The content area
+The content area is the area containing the real content of a layoutable. The properties `width`, `minWidth`, `maxWidth`, `height`, `minHeight` and `maxHeight` control the content size.
+
+### Setting a size
+A layoutable's height and width dimensions determine its size on the screen. The simplest way to set the dimensions of a layoutable is by setting the fixed `width` and `height` property. All dimensions in AsyncDisplayKit's layout system are unitless, and represent density-independent pixels. Setting dimensions this way is common for layoutables that should always render at exactly the same size, regardless of screen dimensions.
+
+```Obj-c
+// Set a height of 0.5 points
+layoutable.width = ASDimensionMake(ASDimensionTypePoints, 0.5);
+
+// Set a width of 50%
+layoutable.width = ASDimensionMake(ASDimensionTypePercentage, 0.5);
+```
+
+Besides width and height there is also `minWidth`, `minHeight`, `maxWidth` and `maxHeight` properties. For example an layoutable to which `minWidth` is applied will never be narrower than the minimum width specified, but it will be allowed to grow normally if its content exceeds the minimum width set. For a full description look at the `Layout Properties` section.
+
+### Sizing Dimensions
+Setting dimensions on layoutables are of the type `ASDimension`. `ASDimension` is essentially a normal CGFloat with support for representing either a point value, or a % value. It allows the same API to take in both fixed values, as well as relative ones:
+
+```Obj-c
+/**
+ * A dimension relative to constraints to be provided in the future.
+ * A RelativeDimension can be one of three types:
+ *
+ * "Auto" - This indicated "I have no opinion" and may be resolved in whatever way makes most sense given the circumstances.
+ *
+ * "Points" - Just a number. It will always resolve to exactly this amount.
+ *
+ * "Percent" - Multiplied to a provided parent amount to resolve a final amount.
+ */
+typedef NS_ENUM(NSInteger, ASDimensionType) {
+  /** This indicates "I have no opinion" and may be resolved in whatever way makes most sense given the circumstances. */
+  ASDimensionTypeAuto,
+  /** Just a number. It will always resolve to exactly this amount. This is the default type. */
+  ASDimensionTypePoints,
+  /** Multiplied to a provided parent amount to resolve a final amount. */
+  ASDimensionTypeFraction,
+};
+```
```

Comment:
Rather than providing a header dump, add each case as a title and write a little bit about its utility. ie.

**ASDimensionTypeAuto**
This is the default dimension. When defined, it allows the layoutable to define the dimension that will fit the best in the current layout context. For example, a `height` set to `Auto` when a specific dimension is defined for minHeight and maxHeight will ensure the layoutable's height is never fixed.